src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.ts

```diff
@@ -1,31 +1,89 @@
-import { Component, Input } from '@angular/core'
-import { KeyBinding } from '../../models/key-binding'
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core'
 import { KeyBindingMatcher } from '../../abstractions/key-binding-matcher.service'
-import { KeyEventType } from '../../value-objects/key-event-type'
+import { Logger } from '../../../core/abstractions/logger.service'
+import { StyledKeyBinding } from '../../value-objects/styled-key-binding'
+import { KeyAliasService } from '../../abstractions/key-alias-service'
+import { PhysicalKeyboardLayoutService } from '../../services/physical-keyboard-layout.service'
+import { KeyboardLayout, KeyboardLayoutKey } from '../../value-objects/keyboard-layout'
 
 @Component({
   selector: 'sofie-virtual-keyboard',
   templateUrl: './virtual-keyboard.component.html',
   styleUrls: ['./virtual-keyboard.component.scss'],
 })
-export class VirtualKeyboardComponent {
+export class VirtualKeyboardComponent implements OnChanges {
   @Input()
-  public keyBindings: KeyBinding[]
+  public keyBindings: StyledKeyBinding[]
```

Comment:
The only difference between `KeyBinding` and `StyleKeyBinding` is that `StyledKyeBinding` has an optional `background` attribute.
Shouldn't this be `backgroundColor` since you store the color of the background in it.

Reply:
The background can besides a solid color, be an image or a gradient.